### PR TITLE
Milestone 3/4

### PR DESCRIPTION
**Pull request by mgunduz1**
Our project contains all requirements in the project specifics. Also we added some extra features to the user/show page. 

> 
> As a guest user:
> 
> I want to be able to create account/log in.
> I can see only “Sign in” and “Sign out” page.
> 
> As a logged-in user:
> 
> I want to be able to see all users list.
> I want to be able to see selected user page with their user name and all posts written by them (the most recent posts on the top).
> I want to be able to send a friendship invitation.
> I want to see a button “Invite to friendship” next to the name of user who is not my friend yet - on both users’ list and single user page.
> I want to be able to see pending friendship invitations sent to me from other users.
> I want to be able to accept or reject friendships invitation.
> I want to be able to create new posts (text only).
> I want to be able to like/dislike posts (but I can like single post only once).
> I want to be able to add comments to posts.
> I want to be able to see “Timeline” page with posts (with number of likes and comments) written by me and all my friends (the most recent posts on the top).
> Timeline page should be the root page of the app.
>

### CODE REVIEW

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
# Status: Change Requested 🤏🤏

Amazing job guys👏👏 @mgunduz1 and @MarkoNS1990 your social media app is almost done because now:

- User can confirm a friend request or reject it
- User can see a list of his/her friends
- User can see posts of his friends by the most recent
- PR is professional

### However, you need to write unit and integration tests so that your project can be approved go ahead and do it and submit another CRR

In case you need assistance you can reach out to me via:
Slack: Delice Lydia | [Linkedin](Linkedin.com/in/delice-lydia/) | [Twitter](https://twitter.com/IngabireLydia3)

## Happy Coding 👐

File: spec/comment_spec.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+require 'rails_helper'
+
+RSpec.describe Comment, type: :model do
+  it 'should belong to user' do
+    should belong_to(:user)
+  end
+
+  it 'should belong to post' do
+    should belong_to(:post)
+  end
```
Comment (enaburekhan) [inline spec/comment_spec.rb:10]: There should be one `expect` statement for each `it` block. The highlighted codes can be rewritten with an `expect` statement using a `context` method like so
```
context 'Association Test' do
    it 'belongs to users' do
      expect(Comment.reflect_on_association(:user).macro).to be :belongs_to
    end

    it 'belongs to post' do
      expect(Comment.reflect_on_association(:post).macro).to be :belongs_to
    end
  end
```

File: spec/friendship_spec.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+    it 'should validate' do
+      should validate_presence_of(:friend_id)
+    end
+  end
+
+  describe 'active record associations' do
+    it { should belong_to(:user) }
+    it { should belong_to(:friend) }
+  end
```
Comment (enaburekhan) [inline spec/friendship_spec.rb]: Include an `expect` statement for each `it` block as already described above.

File: spec/like_spec.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+require 'rails_helper'
+
+RSpec.describe Like, type: :model do
+  describe 'active record associations' do
+    it { should belong_to(:post) }
+  end
+
+  describe 'active record associations' do
+    it { should belong_to(:user) }
+  end
```
Comment (enaburekhan) [inline spec/like_spec.rb]: Include an `expect` statement for each `it` block as already described above.

File: spec/post_spec.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    it 'should validate' do
+      should validate_presence_of(:content)
+    end
+  end
+
+  context 'active record associations' do
+    it { should have_many(:comments) }
+    it { should belong_to(:user) }
+    it { should have_many(:likes) }
+  end
```
Comment (enaburekhan) [inline spec/post_spec.rb]: Include an `expect` statement for each `it` block as already described above.

**Review by enaburekhan — CHANGES_REQUESTED** [review]
# Status: Changes required:wrench::hammer::full_moon:
**Hi team!
Great job so far:clap::clap:. We are almost there. However, we just need to make a few changes to get the project approved:muscle:**

**To improve**

- [ ] Currently, all the tests only cover the model. We also need to write tests for other features implemented. Such as the controllers, etc. Kindly also write tests to cover these features.

- [ ] Check my comments below and make the required changes.

Make the required changes and when you are ready, request another review.

Do have questions/comments? kindly drop them :point_down: and ping me on slack @ Eric Enaburekhan to respond.
Happy hacking:blush:
Eric Enaburekhan.

**Review by elisha2kyakpo1 — APPROVED** [review]
# Status: Approved✔🎉💪
**Congratulations, I'm happy that you have implemented all the changes requested by other reviewers**

You can merge and move.

Cheers🎉🎉
Elisha kyakopo

**Pull request merged**